### PR TITLE
Xenothumbs Minor Event Button Comeback

### PR DESCRIPTION
**Pull request by PhantomEpicness**
# About the pull request
Working version that is TMable for april fools. 

Revived and fixed up https://gitlab.com/cmdevs/colonial-warfare/-/merge_requests/1915/diffs#9ec310cca070259d82839a4a70e89dd858ded438_958_958

https://github.com/cmss13-devs/cmss13/pull/2997

Additions from  Vanagandr's original MR
- xenos can use health items on marines
- vendors are usable now, all checks are simply skipped so introducing xenos with thumbs to snowflake vendors is a horrible idea (as if enabling this in the first place wasn't bad enough) [needs further testing]
- xenos can drive
- M2C usable

I think this is pretty much done, it's already exceeding the scope of the original and we're just making xenos human past this point

Original description
> TL;DR: https://streamable.com/rlvs86
> 
> Mainly this allows xenos to pick up items. They can then use most of them.
> 
> Ex. a xeno can't flip a plasteel cade open - but if she has a toolbelt and time to work, she can unwrench it.
> 
> They can fumble about but a lot of QoL and objects (middleclick container open, click-dragging containers, tables, etc.) were never designed to be interacted with by xenos so item interactions are a bit clunky.
> 
> They can't strip clothing and gear from people.
> 
> They *can* strip and attach attachments but can't aim down scopes.
> 
> They *can* FF.
> 
> They *can't* use MGs.
> 
> IFF works, roughly, not picking up on specific hives but allowing them to shoot past other xenos and hit humans.
> 
> They pass spec checks and can clumsily handle medic gear.
> 
> needless to say this is all colossally imbalanced and has potential for fuckery limited only by the xeno's imagination and whether the thing they're messing with makes an explicit ishuman() check. Or is a structure. They can't hand-interact with structures, so they can shove shells into a mortar but can't change its coords.
> 
> Cosmetically, gun inhands don't line up with xenos. Tweaked it a bit to center it horizontally but lining inhands up with hands for every xeno and xenonid sprite is a bit more hassle than seems worthwhile ATM.
> 

# Explain why it's good for the game
TM funny
# Testing Photographs and Procedure
<details>
<summary>Screenshots & Videos</summary>

Put screenshots and videos here with an empty line between the screenshots and the `<details>` tags.

</details>

# Changelog
:cl:  Vanagandr, totalepicness
add: Added a button to event panel to give xenomorphs opposable thumbs and firearms permits, to individuals, to hives, or to all xenos everywhere. They're clumsy, though, and have issues with fine manipulation sometimes. Remember that warriors cannot throw objects and make for poor grenadiers.
add: Xenos with opposable thumbs can use buckles, drive vehicles, use machine guns and vendors without care for access or skills
/:cl:

### CODE REVIEW

**Review by harryob — CHANGES_REQUESTED** [review]

File: code/game/machinery/vending/cm_vending.dm [hunk 1]
```diff
@@ -508,7 +516,12 @@ GLOBAL_LIST_EMPTY(vending_products)
 	if(.)
 		return
 
-	var/mob/living/carbon/human/user = usr
+	var/mob/living/carbon/human/human_user
+	var/mob/living/carbon/user = usr
```
Comment (hry-gh) [inline code/game/machinery/vending/cm_vending.dm:520]:
```suggestion
	var/mob/living/carbon/user = ui.user
```

File: code/modules/cm_marines/smartgun_mount.dm [hunk 2]
```diff
@@ -70,7 +70,7 @@
 	return
 
 /obj/item/device/m56d_gun/attackby(obj/item/O as obj, mob/user as mob)
-	if(!ishuman(user))
+	if(!ishuman(user) || !HAS_TRAIT(user,TRAIT_OPPOSABLE_THUMBS))
```
Comment (hry-gh) [inline code/modules/cm_marines/smartgun_mount.dm:73]:
```suggestion
	if(!ishuman(user) || !HAS_TRAIT(user, TRAIT_OPPOSABLE_THUMBS))
```

File: code/modules/projectiles/gun_helpers.dm [hunk 3]
```diff
@@ -782,7 +782,7 @@ DEFINES in setup.dm, referenced here.
 	if(flags_gun_features & GUN_BURST_FIRING)
 		return
 
-	if(!ishuman(usr))
+	if(!ishuman(usr) && !HAS_TRAIT(usr, TRAIT_OPPOSABLE_THUMBS))
```
[the remark this answers is not in the document]
Comment (hry-gh) [reply code/modules/projectiles/gun_helpers.dm:785]: bump, just 

```suggestion
	if(!ishuman(user) && !HAS_TRAIT(user, TRAIT_OPPOSABLE_THUMBS))
```

File: code/modules/projectiles/gun_helpers.dm [hunk 4]
```diff
@@ -460,9 +460,9 @@ DEFINES in setup.dm, referenced here.
 		else attack_verb = list("slashed", "stabbed", "speared", "torn", "punctured", "pierced", "gored") //Greater than 35
 
 /obj/item/weapon/gun/proc/get_active_firearm(mob/user, restrictive = TRUE)
-	if(!ishuman(usr))
-		return
 	if(user.is_mob_incapacitated() || !isturf(usr.loc))
+		return
+	if(!ishuman(usr) && !HAS_TRAIT(user, TRAIT_OPPOSABLE_THUMBS))
```
Comment (hry-gh) [inline code/modules/projectiles/gun_helpers.dm:465]:
```suggestion
	if(!ishuman(user) && !HAS_TRAIT(user, TRAIT_OPPOSABLE_THUMBS))
```

File: code/modules/vehicles/interior/interactable/seats.dm [hunk 5]
```diff
@@ -108,7 +108,9 @@
 				to_chat(user, SPAN_WARNING("You are unable to use heavy weaponry."))
 			return
 
-	for(var/obj/item/I in user.contents) //prevents shooting while zoomed in, but zoom can still be activated and used without shooting
+		else if(!HAS_TRAIT(user, TRAIT_OPPOSABLE_THUMBS))
```
Comment (hry-gh) [inline code/modules/vehicles/interior/interactable/seats.dm:111]:
```suggestion
		if(!HAS_TRAIT(user, TRAIT_OPPOSABLE_THUMBS))
```

**Review by hry-gh — CHANGES_REQUESTED** [review]
some more minor nits. it'd be nice to get this actually merged so we don't have to do this next year, given it's behind a verb

File: code/modules/projectiles/gun_helpers.dm [hunk 6]
```diff
@@ -782,7 +782,7 @@ DEFINES in setup.dm, referenced here.
 	if(flags_gun_features & GUN_BURST_FIRING)
 		return
 
-	if(!ishuman(usr) && !HAS_TRAIT(usr, TRAIT_OPPOSABLE_THUMBS))
+	if(!ishuman(user) && !HAS_TRAIT(user, TRAIT_OPPOSABLE_THUMBS))
```
Comment (hry-gh) [inline code/modules/projectiles/gun_helpers.dm:785]:
```suggestion
	if(!ishuman(usr) && !HAS_TRAIT(usr, TRAIT_OPPOSABLE_THUMBS))
```

**Review by hry-gh — CHANGES_REQUESTED** [review]
mb

**Review by harryob — APPROVED** [review]
